test/test.js

```diff
@@ -165,56 +171,51 @@ async function deregisterHost(address = hostAddress, secret = hostSecret) {
+    host.on(evernode.HostEvents.AcquireLease, async (r) => {
+        console.log(`Host received acquire request: '${r.payload}'`);
 
-        // Setup host to watch for incoming redeems.
-        const host = await getHostClient();
+        if (scenario !== "timeout") {
+            console.log(`Host submitting ${scenario} response...`);
+            await new Promise(resolve => setTimeout(resolve, 4000));
 
-        host.on(evernode.HostEvents.Redeem, async (r) => {
-            console.log(`Host received redeem request: '${r.payload}'`);
+            if (scenario === "success")
+                await host.acquireSuccess(r.acquireRefId, r.tenant, { content: "dummy success" });
+            else if (scenario === "error") {
+                const nft = (await (new evernode.XrplAccount(r.tenant)).getNfts())?.find(n => n.TokenID == r.nfTokenId);
+                const leaseIndex = Buffer.from(nft.URI, 'hex').readUint16BE(evernode.EvernodeConstants.LEASE_NFT_PREFIX_HEX.length);
 
-            if (scenario !== "timeout") {
-                console.log(`Host submitting ${scenario} response...`);
-                await new Promise(resolve => setTimeout(resolve, 4000));
-
-                if (scenario === "success")
-                    await host.redeemSuccess(r.redeemRefId, userAddress, { content: "dummy success" });
-                else if (scenario === "error")
-                    await host.redeemError(r.redeemRefId, userAddress, "dummy_error");
+                await host.burnOfferLease(r.nfTokenId);
+                await host.createOfferLease(leaseIndex, r.leaseAmount, tosHash);
+                await host.acquireError(r.acquireRefId, r.tenant, r.leaseAmount, "dummy_error");
             }
-
-            if (++tasks === 2)
-                resolve();
-        })
-
-        await fundUser(user);
-
-        try {
-            const timeout = (scenario === "timeout" ? 10000 : 30000);
-            const result = await user.redeem(hostToken, hostAddress, 1, "dummy request", { timeout: timeout });
-            console.log(`User received instance '${result.instance}'`);
-        }
-        catch (err) {
-            console.log("User recieved redeem error: ", err.reason)
         }
-
-        if (++tasks === 2)
-            resolve();
     })
+
+    await fundTenant(tenant);
+
+    try {
+        const timeout = (scenario === "timeout" ? 10000 : 30000);
+        const result = await tenant.acquire(hostAddress, "dummy request", { timeout: timeout });
+        console.log(`Tenant received instance '${result.instance}'`);
+    }
+    catch (err) {
+        console.log("Tenant recieved acquire error: ", err.reason)
+    }
 }
 
 //////////////////////////////////////////////////////////////////////////////////////
 
-async function getUserClient() {
-    const client = new evernode.UserClient(userAddress, userSecret);
+async function getTenantClient() {
+    const client = new evernode.UserClient(tenantAddress, tenantSecret);
```

Comment:
Let's rename the "UserClient" class to "TenantClient". Are we doing it in separate PR?